spec/features/items/index_spec.rb

```diff
@@ -0,0 +1,58 @@
+require "rails_helper"
+
+RSpec.describe "items#index" do
+  before(:each) do
+    @supermarket_1 = Supermarket.create!(name: "Kroger", location: "555 Food Ave")
+
+    @customer_1 = Customer.create!(name: "Fred")
+    @customer_2 = Customer.create!(name: "Creedence")
+    @customer_3 = Customer.create!(name: "Jane")
+
+    @item_1 = @supermarket_1.items.create!(name: "Eggs", price: 6)
+    @item_2 = @supermarket_1.items.create!(name: "Banana", price: 1)
+    @item_3 = @supermarket_1.items.create!(name: "Milk", price: 4)
+    @item_4 = @supermarket_1.items.create!(name: "Walnuts", price: 5)
+
+    @eggs_1 = CustomerItem.create!(customer: @customer_1, item: @item_1)
+    @eggs_2 = CustomerItem.create!(customer: @customer_2, item: @item_1)
+    @banana = CustomerItem.create!(customer: @customer_1, item: @item_2)
+    @milk_1 = CustomerItem.create!(customer: @customer_1, item: @item_3)
+    @milk_2 = CustomerItem.create!(customer: @customer_3, item: @item_3)
+    # @walnuts = CustomerItem.create!(customer: @customer_1, item: @item_4)
+  end
+    # Story 3
+
+    # As a visitor,
+    # When I visit the items index page,
+    # Then I see a list of all items
+    # including the item's name, price, and the name of the supermarket that it belongs to
+    # and the count of customers that bought that item.
+
+    it "displays all items and their attributes" do
+      visit "/items"
+
+      expect(page).to have_content(@item_1.name)
+      expect(page).to have_content(@item_1.price)
+      expect(page).to have_content(@item_1.supermarket.name)
+      expect(page).to have_content(@item_1.customer_count)
```

Comment:
The structure of this test is excellent -- my only suggestion would be to not call the same customer_count method that you are testing.

Consider the possibility that your logic in the customer_count method is somehow incorrect (it isn't, but let's imagine :) ). If you were to call this same method in the test, then your test would pass because it would be calling this incorrect method in order to verify that the method works. So, a more meaningful assertion is to do the calculation manually (it's pretty easy in this case - Item 1 is only associated with 2 customers) and use that result in the assertion directly: expect(page).to have_content("Count: 2")

That way, we can be sure that the functionality of the method is behaving correctly.